crates/bevy_pbr/src/material.rs

```diff
@@ -210,11 +209,27 @@ impl<M: SpecializedMaterial> Plugin for MaterialPlugin<M> {
                 .add_render_command::<AlphaMask3d, DrawMaterial<M>>()
                 .init_resource::<MaterialPipeline<M>>()
                 .init_resource::<SpecializedMeshPipelines<MaterialPipeline<M>>>()
+                .add_system_to_stage(RenderStage::Extract, extract_materials::<M>)
                 .add_system_to_stage(RenderStage::Queue, queue_material_meshes::<M>);
         }
     }
 }
 
+fn extract_materials<M: SpecializedMaterial>(
+    mut commands: Commands,
+    query: Query<(Entity, &ComputedVisibility, &Handle<M>)>,
+    mut prev_len: Local<usize>,
+) {
+    let mut materials = Vec::with_capacity(*prev_len);
+    for (entity, computed_visibility, material) in query.iter() {
+        if computed_visibility.is_visible {
+            materials.push((entity, (material.clone(),)));
```

Comment:
Does this need to be a strong clone? Each of these calls sends message down a mpsc channel. If we can get away with doing a weak clone or just using the HandleId, it'd likely be much faster.

Reply:
Done. This took it down to 0.267ms + 0.227ms.